### PR TITLE
T2696

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/2696-crash-uncaught-typeerror-cannot-read-property-width-of-undefined

### CODE REVIEW

File: templates/assets/editors/googlemap/googlemap-editor.js [hunk 1]
```diff
@@ -178,34 +179,36 @@
...
-          });
-        });
+        trackEvent.listen( "trackeventupdated", onTrackEventUpdated );
 
         // Now we REALLY know that we can try setting up listeners
-        popcorn.on( "googlemaps-loaded", function() {
-          popcorn.off( "googlemaps-loaded" );
-          getMapFromTrackEvent();
-        });
+        _popcorn.on( "googlemaps-loaded", mapLoaded );
```
Comment (ScottDowne) [inline templates/assets/editors/googlemap/googlemap-editor.js:205]: Shouldn't you also do _popcorn.off in the close function?

File: templates/assets/editors/googlemap/googlemap-editor.js [hunk 2]
```diff
@@ -178,34 +179,36 @@
...
+      _popcorn.off( "googlemaps-loaded", mapLoaded );
+      getMapFromTrackEvent();
+    }
+
+    function onTrackEventUpdated( e ) {
+      _this.updatePropertiesFromManifest( e.target );
+      _this.setErrorState( false );
+
+      // Now we REALLY know that we can try setting up listeners
+      _popcorn.on( "googlemaps-loaded", mapLoaded );
```
Comment (ScottDowne) [inline templates/assets/editors/googlemap/googlemap-editor.js:192]: Do we actually need this here?

Comment (mjschranz) [inline templates/assets/editors/googlemap/googlemap-editor.js:192]: Trust me, we do.

File: templates/assets/editors/googlemap/googlemap-editor.js (hunk 1 above)
Comment (mjschranz) [inline templates/assets/editors/googlemap/googlemap-editor.js:205]: No. The google-maps loaded event is something that is always fired from our plugin. My listener function then handles removing the listener for it.

Comment (ScottDowne) [inline templates/assets/editors/googlemap/googlemap-editor.js:205]: Ah, I see it now.

Very good, sir.

File: templates/assets/editors/googlemap/googlemap-editor.js (hunk 2 above)
Comment (ScottDowne) [inline templates/assets/editors/googlemap/googlemap-editor.js:192]: Putting this here is fine now that I know they remove themselves.

**Pull request merged**